### PR TITLE
Added missing forms from UltimaSoul's FormID system.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
I took the liberty of adding all the forms included currently in @ultima-soul 's ID system that are missing in PE, to at least starting to fill whatever's missing.
(In draft for now, while I test all of them.)

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Ready for review

File: src/data/text/species_names.h [hunk 1]
```diff
@@ -822,93 +810,318 @@ const u8 gSpeciesNames[][POKEMON_NAME_LENGTH + 1] = {
...
+    [SPECIES_MEOWTH_ALOLAN] = _("Meowth"),
+    [SPECIES_PERSIAN_ALOLAN] = _("Persian"),
+    [SPECIES_GEODUDE_ALOLAN] = _("Geodude"),
+    [SPECIES_GRAVELER_ALOLAN] = _("Graveler"),
+    [SPECIES_GOLEM_ALOLAN] = _("Golem"),
+    [SPECIES_GRIMER_ALOLAN] = _("Grimer"),
+    [SPECIES_MUK_ALOLAN] = _("Muk"),
+    [SPECIES_EXEGGUTOR_ALOLAN] = _("Exeggutor"),
+    [SPECIES_MAROWAK_ALOLAN] = _("Marowak"),
+    // Cosplay Pikachu
```
Comment (ultima-soul) [inline src/data/text/species_names.h]: Add newline here.

File: src/pokemon.c [hunk 2]
```diff
@@ -1153,6 +1122,319 @@ const u16 gSpeciesToNationalPokedexNum[NUM_SPECIES] = // Assigns all species to
 	SPECIES_TO_NATIONAL(ZERAORA),
 	SPECIES_TO_NATIONAL(MELTAN),
 	SPECIES_TO_NATIONAL(MELMETAL),
+
+    // Megas
+	[SPECIES_VENUSAUR_MEGA - 1] = NATIONAL_DEX_VENUSAUR,
```
Comment (ultima-soul) [inline src/pokemon.c]: Shouldn't use tabs for this array.

**Review by ultima-soul — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

File: src/data/text/species_names.h [hunk 3]
```diff
@@ -822,93 +810,319 @@ const u8 gSpeciesNames[][POKEMON_NAME_LENGTH + 1] = {
...
+    // Meloetta
     [SPECIES_MELOETTA_PIROUETTE] = _("Meloetta"),
+    // Genesect
+    [SPECIES_GENESECT_DOUSE_DRIVE] = _("Genesect"),
+    [SPECIES_GENESECT_SHOCK_DRIVE] = _("Genesect"),
+    [SPECIES_GENESECT_BURN_DRIVE] = _("Genesect"),
+    [SPECIES_GENESECT_CHILL_DRIVE] = _("Genesect"),
+    // Greninja
+    [SPECIES_GRENINJA_BATTLE_BOND] = _("Giratina"),
+    [SPECIES_GRENINJA_ASH] = _("Giratina"),
```
Comment (ultima-soul) [inline src/data/text/species_names.h]:
```suggestion
    [SPECIES_GRENINJA_BATTLE_BOND] = _("Greninja"),
    [SPECIES_GRENINJA_ASH] = _("Greninja"),
```

**Review by ultima-soul — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Ready for re-re-review

**Review by ultima-soul — APPROVED** [review]

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**